utils/_context/_scenarios/__init__.py

```diff
@@ -106,6 +109,8 @@ def __init__(
             raise ValueError("rc_api_enabled requires use_proxy")
 
         self._required_containers: list[TestedContainer] = []
+        self._external_containers: list[TestedContainer] = []
```

Comment:
Those names does not make very clear the intention of those lists. On the other hands, I see some issue. so let's fix them before finding good names.

Reply:
The idea was basically:

- **required**: this was the existing name and I think it still makes sense as these are the images that _must_ exist. Another name could be "system containers" maybe.
- **external**: this is basically any image that is used by tests but not owned by us, for example databases, message queues, etc. I also thought about calling this "service containers".
- **internal**: these are the containers used by the tests. I called them internal because they are created by this library, this they are not from external sources.

Basically, the distinction is that required containers are the containers for the test framework which have to be started sequentially, external containers are external service containers that can be started in parallel since they are unrelated, and internal containers are the containers for the tests that use both the required and external containers.

Reply:
Ok I understand. Though, I think it's not a good choice. Even if this naming describes a reality, this reality is not what justifies the presence of these three different lists. The real purpose of having three lists is the startup sequence. 

Let say one day somebody adds an "external" container, but for whatever the reason, it needs the proxy. He'll need to find another naming (after having a hard time to understand why it's not working).

The naming must reflect that startup order (and comments  must make obvious what a short name can't explain).

Reply:
An external container by nature would never need the proxy since the proxy is internal so an external container would never know about it, hence the way it's split.

Reply:
> An external container by nature would never need the proxy since the proxy is internal so an external container would never know about it

I disagree, we don't know what will be the future. We can totally imagine an external container sending telemetry metrics we'd like to observe. But whatever, it's only an example, it  changes nothing to the underlying reason : the only reason for those list is the startup order, this point must be reflected in their names.